### PR TITLE
add some arithmetic prim ops

### PR DESCRIPTION
**Pull request by tranma**

### CODE REVIEW

**Comment by jystic** [comment]
@charleso will be disappointed there's no more casting :) /cc @nhibberd

File: src/Icicle/Avalanche/ToJava.hs [hunk 1]
```diff
@@ -280,10 +283,15 @@ primTypeOfPrim p
...
-  cast      M.PrimCastStringOfDouble = Function "Double.toString"
+  toInt    M.PrimToIntFloor      = Function "Math.floor"
+  toInt    M.PrimToIntCeiling    = Function "Math.ceil"
+  toInt    M.PrimToIntRound      = Function "Math.round"
+  toInt    M.PrimToIntTruncate   = Function "Math.trunc"
+
+  toDouble M.PrimToDoubleFromInt = Function "(double)"
+
+  toString M.PrimToStringFromInt    = Function "Integer.toString"
+  toString M.PrimToStringFromDouble = Function "Double.toString"
```
Comment (jystic) [inline src/Icicle/Avalanche/ToJava.hs]: We should probably just delete this, I don't think we're ever going to need it, thoughts @amosr  ?

Comment (tranma) [inline src/Icicle/Avalanche/ToJava.hs]: yeah, maintaining it is an annoyance. But then how are we going to be Enterprise?

File: test/cli/repl/t07-possiblies/script [hunk 2]
```diff
@@ -1,12 +1,12 @@
 -- Divisions return Possibles.
 -- Try with group:
-feature salary ~> group (int (double value / 200)) ~> sum (double value / 5) / 5
+feature salary ~> group (trunc (double value / 200)) ~> sum (double value / 5) / 5
```
Comment (jystic) [inline test/cli/repl/t07-possiblies/script]: this is much much nicer

File: src/Icicle/Avalanche/ToJava.hs (hunk 1 above)
Comment (jystic) [inline src/Icicle/Avalanche/ToJava.hs]: @tranma Node.js backend

File: src/Icicle/Avalanche/ToJava.hs [hunk 3]
```diff
@@ -280,10 +283,15 @@ primTypeOfPrim p
...
   dble     M.PrimDoubleSqrt = Function "Math.sqrt"
 
-  cast      M.PrimCastIntOfDouble = Function "(int)"
-  cast      M.PrimCastDoubleOfInt = Function "(double)"
-  cast      M.PrimCastStringOfInt = Function "Integer.toString"
-  cast      M.PrimCastStringOfDouble = Function "Double.toString"
+  toInt    M.PrimToIntFloor      = Function "Math.floor"
+  toInt    M.PrimToIntCeiling    = Function "Math.ceil"
+  toInt    M.PrimToIntRound      = Function "Math.round"
+  toInt    M.PrimToIntTruncate   = Function "Math.trunc"
```
Comment (jystic) [inline src/Icicle/Avalanche/ToJava.hs]: This doesn't exist

Savage

Comment (tranma) [inline src/Icicle/Avalanche/ToJava.hs]: the SO answer was something along the lines of a DecimalFormatter object
image

**Comment by jystic** [comment]
looks great, :gb: on :green_apple:

**Comment by tranma** [comment]
@jystic tfw you project is so well-tested travis gives up on running the tests

File: src/Icicle/Avalanche/ToJava.hs (hunk 1 above)
Comment (amosr) [inline src/Icicle/Avalanche/ToJava.hs]: concur - kill the java!

**Pull request merged**